### DISCUSSION TITLE
How to genotype based on RNA-seq data?

### DISCUSSION THREAD

**Discussion by sr320** [sr320/ceabigr#9] [Q&A]
Does anyone have suggestions on the best means to identify SNPs in RNA-seq data?

**Comment by kubu4** [comment]
Here's a notebook post from when we tackled this for the crab paper:

https://robertslab.github.io/sams-notebook/2021/10/05/SNP-Characterization-C.bairdi-v3.1-Transcriptome-Assembly-and-Day2-DEG-Pooled-Samples-RNAseq-Data.html

The R Project I created for the analysis is probably easier to read/follow in RStudio instead of the "render" I have in that notebook entry linked above:

https://github.com/RobertsLab/code/tree/master/r_projects/sam/20211005_cbai_SNP_stats

**Reply by kubu4** [reply]
Here's overview of that process:

1. Assemble transcriptome using Trinity.
2. Annotate transcriptome (my pipeline utilized TransDecoder and Trinotate).
3. Align reads to transcriptome (used transcriptome, as we didn't have genome for crab).
4. Make variant calls with `bcftools mpileup`.
5. Use that R Project to filter SNPs and look at annotations and other stuff.

**Comment by ksil91** [comment]
I've been looking for ways to do it as well. A lot of guidance online is for human genomes and GATK, but not as much for "nonmodel" species.  Seems like STAR is a recommended aligner for RNAseq reads when you want to call SNPs, but I assume any RNAseq tailored aligner would work.

- This pipeline seems to have a lot of promise (and its a docker container). I'm sure there are a lot of C. virginica vcfs you could use for the "known variants": https://github.com/CRG-CNAG/CalliNGS-NF
- This is a 2016 Palumbi pipeline, where rnaseq reads are just treated like any NGS read. I get the sense bowtie isn't considered best practices at this point. https://github.com/bethsheets/SNPcalling_tutorial

**Comment by laurahspencer** [comment]
This is a messy but fairly complete [notebook](https://nbviewer.org/github/laurahspencer/O.lurida_QuantSeq-2020/blob/master/notebooks/SNP%20calling%20for%20genetic%20analysis.ipynb) in which I call SNPs from RNA-Seq data, then examine relatedness using a few clustering methods.  Not a gold standard, but could be a good resource for code. 

(here is the direct GitHub link: [SNP calling for genetic analysis.ipynb](https://github.com/laurahspencer/O.lurida_QuantSeq-2020/blob/master/notebooks/SNP%20calling%20for%20genetic%20analysis.ipynb))

UPDATE: I apparently wrote a slurm script to call variants from fastq files - [20210310_genotype-RNASeq.sh](https://github.com/laurahspencer/O.lurida_QuantSeq-2020/blob/master/notebooks/20210310_genotype-RNASeq.sh). It's specific to the _O. lurida_ genome, but could be helpful for somebody.  And there may (or may not) be more streamlined code in this notebook [Genetic Analysis from QuantSeq for O. lurida.ipynb](https://github.com/laurahspencer/O.lurida_QuantSeq-2020/blob/master/notebooks/Genetic%20Analysis%20from%20QuantSeq%20for%20O.%20lurida.ipynb)

**Comment by laurahspencer** [comment]
Update:  Here is a slurm script that I successfully ran last week to call SNPs from rnaseq data:  [genotype-from-RNASeq.sh](https://raw.githubusercontent.com/laurahspencer/red-king_RNASeq-2022/main/scripts/genotype-from-RNASeq.sh). Inputs are sorted RNASeq bam files. Requires reference genome,

**Comment by laurahspencer** [comment]
Once your slurm job is done, here's an RMarkdown notebook that I used to examine relatedness, identify siblings, and correlate expression PC axes against genetic PC axes. The primary input is the "filtered-true.vcf", : 
- [06_Genetic-Structure-Analysis.html](https://htmlpreview.github.io/?https://github.com/laurahspencer/O.lurida_QuantSeq-2020/blob/master/notebooks/06_Genetic-Structure-Analysis.html)  
- [06_Genetic-Structure-Analysis.Rmd](https://raw.githubusercontent.com/laurahspencer/O.lurida_QuantSeq-2020/master/notebooks/06_Genetic-Structure-Analysis.Rmd)
